mathoxide-lib/src/array.rs

```diff
@@ -1,15 +1,110 @@
 use crate::storage::Storage;
+use crate::views;
+use crate::views::ArrayView;
 
-pub struct Array<StorageType> {
+use num_traits::Num;
+
+use std::fmt;
+
+pub struct Array<StorageType, ViewType> {
     storage: StorageType,
+    view: ViewType,
+}
+
+impl<T, StorageType, ViewType> Array<StorageType, ViewType>
```

Comment:
Why not `impl<T, StorageType, ViewType> ArrayView for Array<StorageType, ViewType>`?